index/scorch/optimize_knn.go

```diff
@@ -56,22 +59,45 @@ func (o *OptimizeVR) Finish() error {
 					wg.Done()
 				}()
 				for field, vrs := range o.vrs {
-					searchVectorIndex, closeVectorIndex, err := segment.InterpretVectorIndex(field)
+					vectorIndexInterpretImpl, err := segment.InterpretVectorIndex(field)
 					if err != nil {
 						errorsM.Lock()
 						errors = append(errors, err)
 						errorsM.Unlock()
 						return
 					}
+
+					vectorIndexSize := vectorIndexInterpretImpl.Size()
```

Comment:
this code block should be moved to the beginning of the function - the idea is to do this check once for every searcher creation not for every segment